### DISCUSSION TITLE
Vercel with VPN-protected APIs

### DISCUSSION THREAD

**Discussion by ivan-kleshnin** [vercel/vercel#5537] [Ideas]
**Our QA / Staging API is protected by VPN**. This is necessary to protect from commits mistakenly exposing private data, etc.
Alternatives to that like *unique URLs + password* seems weaker. For example, HTTP Auth can be mistakenly disabled at backend.
I believe the majority of big companies have this layer of protection.

The problem is that it seems incompatible with Vercel – all platforms IPs are dynamic by design and it's not possible to whitelist some of them. So a Vercel machine (running outside our VPN) is unable to request our API and do SSR tasks.

I guess some secret/token-based solution can be invented and applied but I personally wouldn't like to do that from scratch risking  security. We don't have time nor resources to reimplement API on Vercel either.

**I'm interested how other people workaround this, cause it sounds like a common case.** 
I initially asked the support team at Vercel but did not get a relevant answer. Maybe my explanations were unclear.
I think many companies would like to adopt Vercel partially but this VPN question stays in the way.

**Comment by daveteu** [comment]
maybe use a $5 nginx proxy in this use case, and whitelist the nginx proxy IP address?

**Reply by istonejoeljonsson** [reply]
Do you have any proxy in mind?
